lib/trino-parquet/src/main/java/io/trino/parquet/AbstractParquetDataSource.java

```diff
@@ -256,12 +279,15 @@ private static List<DiskRange> mergeAdjacentDiskRanges(Collection<DiskRange> dis
             implements ChunkReader
     {
         private final DiskRange range;
+        private final LocalMemoryContext readerMemoryUsage;
         private Slice data;
         private int referenceCount = 1;
 
-        public ReferenceCountedReader(DiskRange range)
+        public ReferenceCountedReader(DiskRange range, AggregatedMemoryContext memoryContext)
         {
             this.range = range;
+            this.readerMemoryUsage = requireNonNull(memoryContext, "memoryContext is null")
```

Comment:
`requireNonNull` is unnecessary in newer java versions if you immediately dereference the variable.